tools/misc/tpm2_checkquote.c

```diff
@@ -65,7 +66,16 @@ static bool verify(void) {
         return false;
     }
 
-    EVP_PKEY_CTX *pkey_ctx = EVP_PKEY_CTX_new(pkey, NULL);
+#if OPENSSL_VERSION_NUMBER >= 0x10101003L
+    if (ctx.halg == TPM2_ALG_SM3_256) {
+        if (!EVP_PKEY_set_alias_type(pkey, EVP_PKEY_SM2)) {
```

Comment:
Do not hide function returns in the conditional statements.

Reply:
@idesai I will commit a incremental patch to fix it.